### PR TITLE
fix: handled exception gracefully and returned proper response

### PR DESCRIPTION
**Pull request by abdullahwaheed**
**Description**

Gracefully handle exception in legacy and CT endpoints for combined order history

[SONIC-788](https://2u-internal.atlassian.net/browse/SONIC-788)

**Merge checklist:**
Check off if complete *or* not applicable:
- [X] Documentation updated (not only docstrings)
- [x] Unit tests added/updated

**Post-merge:**
- [ ] [Backported](https://openedx.atlassian.net/wiki/spaces/COMM/pages/2065367719/Making+a+pull+request+for+a+named+release) to latest and next named releases

### CODE REVIEW

File: commerce_coordinator/apps/frontend_app_ecommerce/views.py [hunk 1]
```diff
@@ -94,18 +94,23 @@ def get(self, request):
...
+                "next": None,
+                "previous": None,
+                "results": sorted(output_orders, key=lambda item: date_conv(item["date_placed"]), reverse=True)
+            }
 
-        return Response(output)
+            return Response(output)
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.error(f'UserOrdersView threw following exception: [{exc}]')
+            return Response(status=HTTP_400_BAD_REQUEST, data='Something went wrong!')
```
Comment (syedsajjadkazmii) [inline commerce_coordinator/apps/frontend_app_ecommerce/views.py:116]: [Needs Confirmation] 

We decided that we will throw overall error if any of the e-commerce system threw error. right? Like we wont show partial information from one of the system.

File: commerce_coordinator/apps/frontend_app_ecommerce/views.py [hunk 2]
```diff
@@ -94,18 +94,23 @@ def get(self, request):
...
+                "count": request.query_params['page_size'],
+                "next": None,
+                "previous": None,
+                "results": sorted(output_orders, key=lambda item: date_conv(item["date_placed"]), reverse=True)
+            }
 
-        return Response(output)
+            return Response(output)
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.error(f'UserOrdersView threw following exception: [{exc}]')
```
Comment (syedsajjadkazmii) [inline commerce_coordinator/apps/frontend_app_ecommerce/views.py:115]: NIT:
```suggestion
            logger.error(f'[UserOrdersView] An error occured while fetching Orders' Data: [{exc}]')
```

File: commerce_coordinator/apps/frontend_app_ecommerce/views.py (hunk 1 above)
Comment (NawfalAhmed) [reply commerce_coordinator/apps/frontend_app_ecommerce/views.py:116]: I thought we send the partial order data 🤔

Comment (abdullahwaheed) [reply commerce_coordinator/apps/frontend_app_ecommerce/views.py:116]: yes, we won't show partial information from one system

File: commerce_coordinator/apps/frontend_app_ecommerce/views.py [hunk 3]
```diff
@@ -94,18 +94,23 @@ def get(self, request):
...
+                "next": None,
+                "previous": None,
+                "results": sorted(output_orders, key=lambda item: date_conv(item["date_placed"]), reverse=True)
+            }
 
-        return Response(output)
+            return Response(output)
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.error(f'[UserOrdersView] An error occured while fetching Order History.\n Data: [{exc}]')
+            return Response(status=HTTP_400_BAD_REQUEST, data='Something went wrong!')
```
Comment (syedsajjadkazmii) [inline commerce_coordinator/apps/frontend_app_ecommerce/views.py:116]: Please add a test for this exception.

**Review by syedsajjadkazmii — APPROVED** [review]
Approval from my side once you add the test.

File: commerce_coordinator/apps/frontend_app_ecommerce/views.py (hunk 3 above)
Comment (abdullahwaheed) [reply commerce_coordinator/apps/frontend_app_ecommerce/views.py:116]: added tests for exception handling

**Pull request merged**